diff --git a/weasel.js b/weasel.js
index 320806e..fdbb1db 100644
--- a/weasel.js
+++ b/weasel.js
@@ -37,7 +37,8 @@ var re = new RegExp('\\b(' + weasels.join('|') + ')\\b', 'gi');
 
 module.exports = function (text, opts) {
   var suggestions = [];
-  while (match = re.exec(text)) {
+  var match;
+  while ((match = re.exec(text))) {
     var weasel = match[0].toLowerCase();
     if (exceptions.indexOf(weasel) === -1 ||
         text.substr(match.index-4, 4) !== 'too ') {
